amd64.c

```diff
@@ -10,9 +10,7 @@ static thread_local long long co_active_buffer[64];
 static thread_local cothread_t co_active_handle = 0;
 static void (*co_swap)(cothread_t, cothread_t) = 0;
 
-#ifdef LIBCO_MPROTECT
```

Comment:
`LIBCO_MPROTECT` is a general purpose mechanism to enforce memory protection, I don't see why it's made to be a toolchain-specific thing. Remember than if `_WIN32` is not defined, these attributes pass over to the Sys-V code.

Reply:
Hello there! Thank you for the feedback!
Do you mean we should put
```
#ifndef LIBCO_MPROTECT
  section(text)
#endif
```
instead?
`section(text)` is toolchain-specific to gcc and clang.
If we kept 
```
#ifndef _MSC_VER
  section(text)
#endif
```
it would work under macOS with MSVC where LIBCO_MPROTECT is not defined. A bit further down, this code here
```
    #ifdef LIBCO_MPROTECT
    VirtualProtect((void*)co_swap_function, sizeof co_swap_function, PAGE_EXECUTE_READ, &old_privileges);
    #endif
```
should be excluded under MSVC/macOS.

Best regards,
JC

Reply:
Something like:

    #if defined(LIBCO_MPROTECT)
        alignas(4096)
    #else
        #if defined(_MSC_VER)
              section(text)
        #endif
    #endif

would be better because:

1. If someone doesn't define `LICO_MPROTECT` while on another OS/Compiler, we don't want their build to fail (assuming that compiler doesn't supports `section()` specifier).
2. It doesn't get rid of the alignment specifier, which seems to have been deleted in your PR.